src/_common/sticker/collect/modal/modal.ts

```diff
@@ -0,0 +1,37 @@
+import { Component } from 'vue-property-decorator';
+import { Api } from '../../../api/api.service';
+import AppLoading from '../../../loading/loading.vue';
+import { BaseModal } from '../../../modal/base';
+import AppStickerCollect from '../collect.vue';
+
+@Component({
+	components: {
+		AppLoading,
+		AppStickerCollect,
+	},
+})
+export default class AppStickerCollectModal extends BaseModal {
+	isLoading = true;
+	balance = 0;
+	stickerCost = 10;
+
+	async mounted() {
+		const payload = await Api.sendRequest('/web/stickers/dash');
+
+		this.balance = payload.balance;
+		this.stickerCost = payload.stickerCost;
+
+		this.isLoading = false;
+	}
```

Comment:
Why are we doing it like this and not just passing these as props to the component?

Reply:
The modal is the thing that grabs the balance/sticker cost. We show it when they're able to get a new sticker, not just on the dashboard page.